src/Test/DepTyCheck/Gen.idr

```diff
@@ -569,10 +530,8 @@ forgetAlternatives g@(OneOf {}) = case canBeNotImmediatelyEmpty em of
   Right _   => single g
   Left Refl => maybe Empty single $ strengthen {em=MaybeEmptyDeep} g
   where
-    %inline single : iem `NoWeaker` MaybeEmptyDeep => iem `NoWeaker` em => Gen iem a -> Gen em a
+    %inline single : (0 _ : iem `NoWeaker` MaybeEmptyDeep) => (0 _ : iem `NoWeaker` em) => Gen iem a -> Gen em a
     single g = label "forgetAlternatives" $ OneOf $ MkGenAlts [(1, g)]
-    -- `mkOneOf` is not used here intentionally, since if `mkOneOf` can be changed to eliminate single-element `MkGenAlts`'s,
-    -- we still want such behaviour here.
```

Comment:
Why was this comment removed? It looks as if it still is applicable to this code.

Reply:
Because this comment has been removed as not planned
https://github.com/buzden/deptycheck/pull/211/files#diff-1d8d773de1ded7619cfa19c0072d3d59923fec8c79da88a537ff672d25f253acL225
Maybe replace constructor with mkOneOf?

Reply:
> Maybe replace constructor with mkOneOf?

No, this would badly reflect the intention of this implementation. We can change the comment to "since `mkOneOf` can potentially change the shape of produced tree, but we still want precisely this shape here." if the current form feels unsatisfying.